src/NuGet.Clients/NuGet.CommandLine/Common/Console.cs

```diff
@@ -252,38 +260,45 @@ public void PrintJustified(int startIndex, string text, int maxWidth)
         {
             if (maxWidth > startIndex)
             {
-                maxWidth = maxWidth - startIndex - 1;
+                maxWidth = maxWidth - startIndex;
             }
 
             lock (_writerLock)
             {
-                while (text.Length > 0)
+                using (StringReader stringReader = new StringReader(text))
                 {
-                    // Trim whitespace at the beginning
-                    text = text.TrimStart();
-                    // Calculate the number of chars to print based on the width of the System.Console
-                    int length = Math.Min(text.Length, maxWidth);
-
-                    string content;
-
-                    // Text we can print without overflowing the System.Console, excluding new line characters.
-                    int newLineIndex = text.IndexOf(Environment.NewLine, 0, length, StringComparison.OrdinalIgnoreCase);
-                    if (newLineIndex > -1)
+                    string line;
+                    while ((line = stringReader.ReadLine()) != null)
                     {
-                        content = text.Substring(0, newLineIndex);
+                        if (line == string.Empty)
+                        {
+                            Out.WriteLine();
+                            continue;
+                        }
+
+                        while (true)
+                        {
+                            // Trim whitespace at the beginning
+                            line = line.TrimStart();
+                            // Calculate the number of chars to print based on the width of the System.Console
+                            int length = Math.Min(line.Length, maxWidth);
+
+                            string content = line.Substring(0, length);
+                            content = content.TrimEnd();
+
+                            int leftPadding = startIndex + content.Length - CursorLeft;
+
+                            // Print it with the correct padding
+                            Out.WriteLine((leftPadding > 0) ? content.PadLeft(leftPadding) : content);
+
+                            if (content.Length == line.Length)
```

Comment:
Just to be on safe side `if (content.Length >= line.Length)` maybe better, otherwise might turn into infinite loop.

Reply:
I'm not sure why doing this would help prevent infinite loops. `content` is a substring of `line`, so at most it will be as long as `line`, and cannot be longer, so I don't believe this change would affect the logic in this method.

Reply:
I think your logic works to eventually make the 2 lengths equal. Though, it is a little confusing. 
I tried a case with spaces and I think your logic will unnecessarily output one line. May want to add a test case for it.

line: **"  abcd    "** (2 spaces start; 4 spaces end) so total length: 10
line: "abcd    " length: 8
content: "abcd" content.length: 4

4 != 8, so... line: "    "
<repeat loop>

line: "    "
line: ""
line.length = 0
content = 0
leftPadding = startIndex + 0 - CursorLeft
leftPadding = (possibly not 0)
Writeline (**lots of padded spaces (leftPadding) **)
0 == 0 so break.

Reply:
I've made changes to address this. I've changed where we trim spaces, and how we choose to break/print an empty line. I've also added new tests for whitespace handling, and the new implementation passes these. Let me know if the logic in the method makes more sense now.